keras/src/backend/jax/trainer.py

```diff
@@ -233,7 +235,10 @@ def concatenate(outputs):
                     return output
 
                 if not self.run_eagerly and self.jit_compile:
-                    concatenate = jax.jit(concatenate)
+                    if is_nnx_backend_enabled():
```

Comment:
Can we put this in a util or something? So we can just call the same fn that resolves correctly.